cardshifter-core/src/main/java/com/cardshifter/core/groovy/EffectDescription.groovy

```diff
@@ -0,0 +1,55 @@
+package com.cardshifter.core.groovy
+
+class EffectDescription {
+
+    /**
+     * The description (value of map) should be cased and punctuated as if in the middle of a sentence. This makes it
+     * possible to later change the ordering of description components. It's easier to capitalize and add punctuation
+     * later than the reverse. A map is useful instead of enum et al because it's runtime extensible. */
+    static Map<String, String> vocabulary = new HashMap()
+
+    // Key into vocabulary
+    String triggerId
+
+    // choose $randomChoiceCount atRandom { $randomChoices... }
+    int randomChoiceCount
+    List<EffectDescription> randomChoices
+
+    private StringBuilder builder = new StringBuilder()
+
+    public static setupStandardVocabulary() {
+        vocabulary.putAll([
+                afterPlay: '',
+                onStartOfYourTurn: 'at the start of your turn',
+                onStartOfOpponentsTurn: 'at the start of the opponent\'s turn',
+                onStartOfAnyTurn: 'at the start of a turn',
+                onEndOfYourTurn: 'at the end of your turn',
+                onEndOfOpponentsTurn: 'at the end of the opponent\'s turn',
+                onEndOfAnyTurn: 'at the end of a turn',
+                onDeath: 'when this dies',
```

Comment:
would be nice if we could have "when X dies". Where "X" can be... well, almost anything. "this", "one of your other creatures", "opponent creature", "a Greek creature", "a Greek creature with less than 5 health" (I don't see why it should not be supported...)